cmd/csalt/main.go

```
@@ -248,6 +281,43 @@ func apiFromArgs(args Args) (*userapi.CacophonyUserAPI, string, error) {
 	return api, saltPrefix, nil
 }
 
+func checkForDuplicates(devices *userapi.DeviceResponse) error {
+	nameMap := make(map[string][]userapi.Device)
+	duplicateNames := make([]string, 0, 1)
+	for _, device := range devices.NameMatches {
+		if _, ok := nameMap[device.DeviceName]; !ok {
+			nameMap[device.DeviceName] = []userapi.Device{device}
+		} else {
+			nameMap[device.DeviceName] = append(nameMap[device.DeviceName], device)
+			duplicateNames = append(duplicateNames, device.DeviceName)
+		}
+	}
+	if len(duplicateNames) > 0 {
+		fmt.Printf("DeviceName Query found %v Duplicate Device please specify group:devicename\n", len(duplicateNames))
+		for _, name := range duplicateNames {
+			fmt.Printf("Device %v matches:\n", name)
+			for _, device := range nameMap[name] {
+				fmt.Printf("%v:%v\n", device.GroupName, device.DeviceName)
+			}
+		}
+		return fmt.Errorf("DeviceName Query found %v Duplicate Device please specify group:devicename\n", len(duplicateNames))
```

This should either return the error, or log but not both otherwise it's likely the same output is shown twice.